Deprecate needsOrg & wantsOrg

RasPhilCo opened the pull request:
ref #126 

Waiting on Deprecation Warning date from Enterprise

Commit:
updated example plugin

Commit:
appveyor run in band

Commit:
build cli-engine if needed

Commit:
build cli-engine if needed

Commit:
simplify babel

Commit:
updated example plugin

Commit:
flow-typed

Commit:
uniq topics

Commit:
update example plugin

Commit:
uniq commands

Commit:
updated example plugin

Commit:
fix building

Commit:
node 6

Commit:
do not detach updater

Commit:
updated cli-engine-command

Commit:
updated example-plugin

Commit:
updated example-plugin

Commit:
updated example-plugin

Commit:
updated deps

Commit:
fix bin name

Commit:
support variableArgs

Commit:
default to null

Commit:
1.1.0

Commit:
find yarn

Commit:
added plugins:update

Commit:
added test stub

Commit:
exit with 127

Commit:
test not found

Commit:
added not found test

Commit:
sha256xz

Commit:
ignore files

Commit:
removed old changelog

Commit:
1.1.1

Commit:
added which command

Commit:
added which tests

Commit:
added p7zip-full

Commit:
update image

Commit:
added windows build

Commit:
run in band

Commit:
run in band

Commit:
1.1.2

Commit:
build x64

Commit:
1.1.3

Commit:
rename platform to win32

Commit:
shorten logs

Commit:
shorten logs

Commit:
hide output

Commit:
show help on topic-less commands

Commit:
do not fully error if having trouble grabbing commands

Commit:
show help

Commit:
distance strings when not found

Commit:
updated submodule url

Commit:
update cli-engine-command

Commit:
update cli-engine-command

Commit:
remove Base

Commit:
updated deps

Commit:
move up yarn

Commit:
done for output

Commit:
use heroku mixin

Commit:
set timeout interval

Commit:
fix eventemitter warning

Commit:
Merge pull request #9 from heroku/refactor-base

remove Base

RasPhilCo committed:
Add needsOrg, fixes #12

RasPhilCo committed:
Add help support, fixes #1

Commit:
updated example-plugin

Commit:
cleaner npm files

Commit:
run in band

Commit:
update cli-engine-command

Commit:
1.2.0

Commit:
add remote flag

Commit:
updated cli-engine-command

Commit:
1.2.1

Commit:
consolidated integration tests

Commit:
increase timeout of test

Commit:
fix up deps

Commit:
Merge pull request #10 from heroku/needs-org

Add needsOrg

Commit:
do not skip

RasPhilCo committed:
remove example

RasPhilCo committed:
add example submodule

RasPhilCo committed:
edit .gitmodules

Commit:
updated deps

Commit:
fixed which tests

Commit:
fixed update

Commit:
fixed things

Commit:
use new flag syntax

Commit:
Merge pull request #17 from heroku/submodule-example

Submodule example

Commit:
update example

Commit:
2.0.0

Commit:
updated cli-engine-command

Commit:
updated example

Commit:
updated example-plugin

Commit:
run yarn

Commit:
add flow-copy-source

Commit:
update example

Commit:
2.0.1

Commit:
skip bumping example

Commit:
2.0.2

RasPhilCo committed:
Remove non cli plugin artifacts on install attempt

RasPhilCo committed:
Merge pull request #20 from heroku/bad-plugin-install

Remove non cli plugin artifacts on install attempt

RasPhilCo committed:
Remove example project deployment

RasPhilCo committed:
2.0.3

RasPhilCo committed:
Add config to legacy context

RasPhilCo committed:
Remove example cache from CI

RasPhilCo committed:
cli-engine-command 2.2.6

RasPhilCo committed:
Merge pull request #27 from heroku/add-config-to-ctx

Add config to legacy context

RasPhilCo committed:
2.0.4

RasPhilCo committed:
help flag fix

RasPhilCo committed:
2.0.5

RasPhilCo committed:
2.0.6

Commit:
2.0.7

RasPhilCo committed:
Let cli determine if cmd is asking for help

RasPhilCo committed:
Merge pull request #34 from heroku/help-rules

Let cli determine if cmd is asking for help

RasPhilCo committed:
2.0.8

RasPhilCo committed:
Add error.log truncation, closes #37

RasPhilCo committed:
pr edits

RasPhilCo committed:
pr edits

RasPhilCo committed:
pr edits

RasPhilCo committed:
pr edits

Commit:
Merge pull request #43 from heroku/truncate-error-log

Add error log truncation, closes #37

RasPhilCo committed:
2.0.9

Commit:
updated deps

Commit:
Fix bug in git:remote & add missing attributes (#55)

Commit:
2.0.10

Commit:
added installing via dist-tags (#25)

* added installing via dist-tags

* added test for install

* testing

* finish dist-tagging

* added plugins:update test

* added test for plugins:index

RasPhilCo committed:
Add V5 plugins migration, closes #13 (#31)

Commit:
move Dockerfile to heroku/cli-engine-docker

Commit:
use new circle 2.0 syntax (#59)

* use new circle 2.0 syntax

* take out slow integration test

* fix build step

* only build linux

* fix circle.yml

Commit:
updated rwlockfile (#57)

Commit:
test/refactor plugin cache (#58)

* test/refactor plugin cache

* circle fixes

* circle caching

Commit:
refactor to use cli-engine-config (#60)

* refactor to use cli-engine-config

* updates

* updated example

Commit:
3.0.0

Commit:
fix run.js script

Commit:
3.0.1

RasPhilCo committed:
update cli-engine-command & cli-engine-config

RasPhilCo committed:
3.0.2

RasPhilCo committed:
Edit migration debug output (#62)

Commit:
Adding builtin commands by absolute path

Commit:
Revert "Adding builtin commands by absolute path"

This reverts commit f1e6a46fe356742be13b75433e1b37e5dcacc807.

RasPhilCo committed:
Add TopicHelpPresenter (#56)

Commit:
Remove unnecessary plugins attribute (#71)

Commit:
Adding builtin commands by absolute path (#66)

Commit:
Move undefault down into the require (#72)

Commit:
Remove the forceSilently option from uninstall (#67)

Commit:
Move {name, version} builtin handling to pjson (#73)

Commit:
Factor out {user,builtin,core} plugins from plugin (#68)

* Factor out {user,builtin,core} plugins from plugin

* Make list() a property & remove path.resolve

* Fix core_plugins comment

Commit:
no update when no plugins (#74)

RasPhilCo committed:
cli-engine-command 3.0.3

RasPhilCo committed:
Fix topic help

RasPhilCo committed:
Migrate/try only once (#65)

Commit:
updated deps

Commit:
3.0.3

Commit:
Extract plugin from plugins (#69)

* Extract classes to plugins directory

* Clean up after bad merge conflict fix

* Fix migrator.js after merging master

* Move yarn.js down into plugins directory

Commit:
default commands (#47)

Commit:
3.0.4

Commit:
Clear cache by path (#70)

* Refactor clearCache to take a path instead of name

* Lock user.test.js along with integration.test.js

* Fix linked.test.js after merging master

Commit:
Add PluginPath and refactor plugins dependency (#76)

Commit:
updated deps

Commit:
only warn on cache file read/parse errors (#49)

* only warn on cache file read/parse errors

Fixes https://github.com/heroku/cli-engine/issues/32

* Send cache error to debug rather than warn

Commit:
3.0.5

RasPhilCo committed:
yarn install if yarn.lock missing (#79)

RasPhilCo committed:
3.0.6

Commit:
fix invalid installs (#82)

* fix invalid installs

* flow fix

KarateCowboy committed:
Analytics (#22)

* worknig commit - squash later

* working commit - squash later

* All tests passing commit

* working commit -- squash later

* remove update test file -- out of scope

* update version of cli-engine-command

* update lock file

* fix analytics stubbing

* Add example back as a git submodule

* Stub out submitAnalytics in updater tests

 - remove unnecessary clearAllMocks in analytics tests

* add netrc-parser module dependency

* fix some flowtype errors

* fix 400 bad request error

* remove errant end paren

* update yarn.lock

* working commit for sharing

* tests passing

* flow and standard updates

* revert test changes around cli-engine-config

Remove additions to test files that stubbed out cli-engine-config
parts. Those stubs are not needed any more

* update version of cli-engine-config

* clean up flow errors

* fix formatting

* review feedback addressed

* minor changes for passing build

* make analytics work

* set valid flag

* fix standard

* updated http-call

* Add tests & refactor analytics.js a little (#83)

* Add tests & refactor analytics.js a little

* Rename platform to os

* Rename pluginVersion to plugin_version

* Add language to analytics command

* Remove unused arch attribute from the json

* Fix type definition + standard

* Use vars.apiHost instead of api.heroku.com

Commit:
3.0.7

Commit:
fix cli_binpath (#84)

Commit:
3.0.8

Commit:
update to client directory not cli (#85)

Commit:
3.0.9

Commit:
run in band (#89)

Commit:
skip migration when no user plugins (#80)

Commit:
Use tar-fs instead of tar-stream (#87)

Commit:
Move node.exe when it exists & cleanup next update (#91)

* Move node.exe when it exists & cleanup next update

* Fix standard error

* Use renameSync on windows / moveSync on rest

* Add comments to updater.js

* Add catches to directory operations

* Wrap client.DELETE removal in catch

* Move node.DELETE into tmp directory

* tmp/client.DELETE

* standard fixes

Commit:
clean up yarn.lock

Commit:
3.0.10

Commit:
create tmp dir if not exist

Commit:
3.0.11

Commit:
Add cwd param to yarn & remove yarn.info (#98)

Commit:
updated deps

Commit:
flow-typed

Commit:
Validate that download matches manifest sha (#102)

Commit:
use yarn from file instead of node_modules (#92)

* use yarn from file instead of node_modules

* added TODO

* bump docker version

* remove roadrunner cache hack

* uglify and things

* Add error handling and tmp download (#93)

* Move the yarn download from lib to yarn (#95)

* Move yarn version to package.json

* Move yarn dep down into 'cli-engine' attribute

* Move the yarn download from lib to yarn

* fork -> execa

* Yarn file fork merge master (#100)

* Add cwd param to yarn & remove yarn.info (#98)

* Remove no longer used stdout buffering

* remove get-stream and ran flow-typed

Commit:
updated deps

Commit:
3.0.12

Commit:
Refactor into tmp directory for client / extract (#103)

* Refactor into tmp directory for client / extract

* Clean up empty directories asap

Commit:
touch main file when no prepare script (#45)

* touch main file when no prepare script

Fixes https://github.com/heroku/cli-engine/issues/14

* touching main no longer necessary

* solve race condition of linked plugin changing during update loop

* Do not check if we need prepare if installed (#101)

* Do not check if we need prepare if installed

* prepare is done automatically by yarn now

* added touch back in

* cwd set in constructor

* set new date to 0

Commit:
v3.1.0

RasPhilCo committed:
Return if no cmd, fixes #104

RasPhilCo committed:
3.1.1

Commit:
updated cli-engine-command

Commit:
v3.1.2

RasPhilCo committed:
Add namespaces (#90)

RasPhilCo committed:
cli-engine-command 3.1.3 & cli-engine-config 1.1.3

RasPhilCo committed:
Use bin name in help msgs

RasPhilCo committed:
edits

RasPhilCo committed:
edits

RasPhilCo committed:
edits

RasPhilCo committed:
namespace fixes (#108)

Commit:
v3.2.0

RasPhilCo committed:
edits

Commit:
Post install & cli along with commands & user (#109)

Commit:
v3.2.1

RasPhilCo committed:
edits

RasPhilCo committed:
edits

RasPhilCo committed:
3.2.2

RasPhilCo committed:
Guard against bad topics (#114)

RasPhilCo committed:
3.2.3

Commit:
gracefulify (#115)

Commit:
gitignore

Commit:
fixes for homebrew (#122)

Commit:
v3.2.4

Commit:
updated example-plugin

RasPhilCo committed:
Namespace refactor (#124)

Commit:
updated deps

Commit:
refactor help and use custom help formatters (#121)

* made help test a simpler string

* use help generators from cli-engine-command

Commit:
v3.3.0

Commit:
Fix usage for help for legacy plugins (#132)

* Fix usage for help for legacy plugins

* Remove globals comment from help.test.js

Commit:
Add --non-interactive to yarn args (#131)

Commit:
Clear out analytics file on http errors (#128)

Commit:
v3.3.1

Commit:
Refactor Plugins & IPluginsManager (#134)

* Move src/plugins.js -> src/plugins/index.js

* Remove `Plugins` suffix from attributes

* Rename plugin_manager.js -> manager.js

* Rename IPluginManager to Manager

* Change Manager from an interface to class

Commit:
typo

Commit:
fix uninstalling

Commit:
increase timeout for appveyor

Commit:
v3.3.2

Commit:
Fix yarn remove --non-interactive ordering (#137)

Commit:
v3.3.3

RasPhilCo committed:
cli-engine-command 4.0.0 (#139)

RasPhilCo committed:
Add --team flag support (#140)

Commit:
set coverage directory (#143)

so the submodule directory coverage files are not also uploaded

Commit:
updated dev deps

Commit:
flow-typed

RasPhilCo committed:
Deprecate --org, needsOrg & wantsOrg

jdx commented:
we'll need to wait until all the core plugins have moved over to the new flag to use this

jdx commented:
@dmcinnes as part of this, should we allow people to use `--org` in place of `--team` but hide the `--org` flag in help?

Commit:
set debug headers (#142)

* set debug headers

Fixes #111

* show headers if debug > 1

Commit:
Update example-plugin submodule

Commit:
check yarn exit code (#138)

* [wip] check yarn exit code

right now this breaks under `run plugins:update` because `yarn upgrade`
is returning 1 for some reason

* Extract the yarn error & fix unlocking (#141)

RasPhilCo committed:
Truncate long usage (#146)

RasPhilCo committed:
Add v5 flag parsing (#147)

Commit:
fix quoted arguments (#151)

Commit:
v3.3.4

Commit:
Back out flags changes (#152)

* Revert "Add v5 flag parsing (#147)"

This reverts commit 7b647dd3015da7df573105b5d7107c7c0aaa0771.

* Revert "Add --team flag support (#140)"

This reverts commit b1d1283a204f68f5f3f9f1340b6cb43aafd9ff57.

* Revert "cli-engine-command 4.0.0 (#139)"

This reverts commit 81ed73bd61d9dfedebccad0625a77c4381849eb2.

Commit:
v3.3.5

Commit:
Replace download with got to fix redirects (#156)

Commit:
Throw error if uglify fails (#157)

* Throw error when minify fails

* Hard locking uglify to 73d6438 temporarily

Commit:
Speed up integration / namespace tests (#148)

* Speed up integration / namespace tests

* Add tmp cleaning to `clean` task & appveyor

* Add npm-packages-offline-cache to the circle cache

* Force jest to run in band again due to yarn bug

* https://github.com/yarnpkg/yarn/issues/683

* Remove `yarn run clean` from appveyor

Commit:
Bringing in cli-engine-command 4.0.3 to fix equals flag (#154)

* Revert "Back out flags changes (#152)"

This reverts commit 18fc42ea7db32483608f02606306e0cdaeceb77f.

* cli-engine-command 4.0.3

* cli-engine-heroku 0.3.1

Commit:
Fix bug where command not found after migrate (#155)

* Fix bug where command not found after migrate

* Fix migrator test after merging master

Commit:
v3.3.6

jdx commented on src/plugins/legacy.js:
```diff
@@ -88,6 +89,7 @@ export function convertFromV5 (c: LegacyCommand): Class<Command<*>> {
     V5.flags.remote = Flags.remote()
   }
   if (c.needsOrg || c.wantsOrg) {
+    new Output().warn(`--org in ${c.topic}${c.command ? ':' + c.command : ''} is deprecated`)
```
you should be able to use `this.out`. You should never create a new output because it won't reuse the existing config object if you do this

Commit:
Add node-gyp with yarn hack for heroku-kafka build (#160)

* Add node-gyp with yarn hack for heroku-kafka build

* Remove unused Env

Commit:
v3.3.7

Commit:
Remove undefined attributes on legacy flag conversion (#153)

dmcinnes commented:
@dickeyxxx yeah that sounds right to me.

dmcinnes commented:
Whoops thought this was my PR 😅  In any case it's updated.

RasPhilCo commented:
@jdxcode Lol. PR is no good now, can't even see what the original change was. I'll address this if/when they tell us the deprecation warning should go live.
